code/datums/abilities/carnival_cant.dm

```diff
@@ -0,0 +1,36 @@
+/datum/abilityHolder/carnival
+    usesPoints = 0
+    regenRate = 0
+    tabName = "Carnival"
+
+/datum/targetable/carnival/cant //defaults to clown version
+    name = "Carnival Cant"
+    desc = "A secret language that only entertainers know! This one in particular is a form of sign language!"
+    icon_state = "cant-clown"
+    targeted = 0
+    cooldown = 0
+    pointCost = 0
+
+    cast()
+        if(!holder)
+            return TRUE
+        if(!ishuman(holder.owner))
+            return
+        var/mob/living/carbon/human/OWNER = holder.owner
+        if(!OWNER.limbs.l_arm && !OWNER.limbs.r_arm)
+            OWNER.show_text("You can't speak Carnival Cant without arms!","red")
+        var/message = input(OWNER, "What would you like to say?", "Carnival Cant") as null|text
+        if(!length(message) || (copytext(message,1,2) == " "))
+            message = null
+        if(!message)
+            return
+        OWNER.emote("carnivalcant")
+        for(var/mob/LISTENER in viewers(OWNER,10))
```

Comment:
believe you need to use `AIviewers` here since you want the ai to be able to hear this

Reply:
ill test it, they should be able to hear though.

Reply:
yep yep! works perfectly fine～